Apsitvarkom.DataAccess/GoogleGeocoder.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Apsitvarkom.Configuration;
+using Apsitvarkom.Models;
+using System.Text.Json;
+using Yoh.Text.Json.NamingPolicies;
+
+namespace Apsitvarkom.DataAccess;
+
+public class GoogleGeocoder : IGeocoder
+{
+    private readonly HttpClient _httpClient;
+    private readonly string _apiKey;
+
+    private static readonly JsonSerializerOptions SerializerOptions = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicies.SnakeCaseLower
+    };
+
+    public GoogleGeocoder(HttpClient httpClient, IApiKeyProvider apiKeyProvider)
+    {
+        _apiKey = apiKeyProvider.Geocoding;
+        _httpClient = httpClient;
+    }
+
+    public async Task<string?> ReverseGeocodeAsync(Coordinates coordinates)
+    {
+        var jsonStream = await _httpClient.GetStreamAsync($"json?latlng={coordinates.Latitude},{coordinates.Longitude}&language=lt&result_type=political&key={_apiKey}");
```

Comment:
nice catch with the reversed coordinates

Reply:
yeah it was returning UAE when in fact I wanted to get Vingio parkas

Reply:
There's also `language=lt` in the query. I am not sure how we want this handled. We could switch to `en`.

Or maybe we could come up with something more interesting and make the request in both languages (since we're planning to internationalise (add languages) to our front end)

Reply:
internationalization sounds better imo.

Reply:
Let's keep that in mind and maybe we could create a task for that. Although we would need to figure out how to best do it and how it will be saved to the database